Sources/UIView.swift

```diff
@@ -206,7 +206,13 @@ open class UIView: UIResponder, CALayerDelegate, UIAccessibilityIdentification {
     /// Adds a subview without touching the view's layer or any of its sublayers.
     /// We need to be able to add layers at an index that isn't directly related to its subview index.
     private func insertSubviewWithoutTouchingLayer(_ view: UIView, at index: Int) {
-        if view.superview != nil { view.removeFromSuperview() }
+        // remove from superview without removing from superlayer
+        if let oldSuperview = view.superview {
+            oldSuperview.subviews = oldSuperview.subviews.filter { $0 != view }
```

Comment:
should we use `oldSuperview.subviews.removeAll(where: { $0 == view })` maybe?

Reply:
Yeah, seems to make more sense to me this way. Just copied this line from `removeFromSuperview()` below, so we probably want to adjust this there too.